lib/rules/require-camelcase-or-uppercase-identifiers.js

```diff
@@ -113,11 +188,104 @@ module.exports.prototype = {
     },
 
     check: function(file, errors) {
+
+        // Return undefined or the start of the unprefixed value
+        function startAfterStringPrefix(value, prefix) {
+            var start;
+            if (((1 + prefix.length) < value.length) &&
```

Comment:
Instead of one `if` would you mind trying something like 
```js
var start = prefix.length + 1;

if (len >= value.length)) {
  return;
}
...

return start;
...

return start;
```
?